distributed/utils.py

```diff
@@ -386,7 +386,7 @@ def key_split_group(x):
     typ = type(x)
     if typ is tuple:
         return x[0]
-    elif typ is str:
+    elif typ is unicode or typ is str:
```

Comment:
In Python 3 this is redundant.  In Python 2 the bytes case is handled just below.  Would everything work just as well if you dropped the second condition here?  Alternatively we could always have two separate functions that we define based on if we're in Python 2 or Python 3.